tests/integration/Application/CommandHandler/Entity/EntityChangeRequestCommandHandlerTest.php

```diff
@@ -0,0 +1,179 @@
+
+    private RequestStack|Mock $requestStack;
+
+    private LoggerInterface|Mock $logger;
+
+    private TicketService|Mock $ticketService;
+
+    private MailService|Mock $mailService;
+
+    private EntityServiceInterface|Mock $entityService;
+
+    private EntityChangeRequestRepository|mock $entityChangeRequestRepository;
+    public function setUp(): void
+    {
+        $this->entityChangeRequestRepository = m::mock(EntityChangeRequestRepository::class);
+        $this->entityService = m::mock(EntityServiceInterface::class);
+        $this->ticketService = m::mock(TicketService::class);
+        $this->requestStack = m::mock(RequestStack::class);
+        $this->logger = new NullLogger();
+
+        $this->mailService = m::mock(MailService::class);
+
+        $this->commandHandler = new EntityChangeRequestCommandHandler(
+            $this->entityChangeRequestRepository,
+            new ContractualBaseService(),
+            $this->entityService,
+            $this->ticketService,
+            $this->requestStack,
+            $this->mailService,
+            $this->logger,
+            'customIssueType'
+        );
+
+    }
+
+    public static function contractualBaseProvider(){
+        return [
+            'Pristine equals enitity, no change' => [Constants::CONTRACTUAL_BASE_IX, []],
+            'Pristine is null, update to correct value' => [null, ["metaDataFields.coin:contractual_base" => "IX"]],
+            'Pristine is wrong, update to correct value' => [Constants::CONTRACTUAL_BASE_AO, ["metaDataFields.coin:contractual_base" => "IX"]],
+        ];
+    }
+
+    /**
+     * @dataProvider contractualBaseProvider
+     */
+    public function test_it_should_update_contractual_base_on_change_request(?string $pristineContractualBase, array $expectedDiff)
+    {
+        $manageEntity = $this->createEntity();
+
+        $applicant = new Contact('john:doe', 'john@example.com', 'John Doe');
+
+        $pristineEntity = unserialize(serialize($manageEntity));
+        if($pristineContractualBase !== null) {
+            $pristineEntity->getMetaData()->getCoin()->setContractualBase($pristineContractualBase);
+        }
+
+        $this->entityService->shouldReceive('getPristineManageEntityById')
+            ->once()
+            ->with($manageEntity->getId(), $manageEntity->getEnvironment())
+            ->andReturn($pristineEntity);
+
+        $ticket = new Issue('ISSUE-123', 'foo', 'bar');
+
+        $this->ticketService->shouldReceive('createJiraTicket')
+            ->once()
+            ->andReturn($ticket);
+
+        $this->entityChangeRequestRepository->shouldReceive('openChangeRequest')
+            ->once()
+            ->withArgs(function($entity, $pristineEntity) use ($expectedDiff){
+                if($pristineEntity->diff($entity)->getDiff() !== $expectedDiff){
+                    var_dump($pristineEntity->diff($entity)->getDiff(), 'is not equal to' , $expectedDiff);
```

Comment:
:thinking: is this a dev leftover? Can you not do a $this->fail? or throw an exception?

Reply:
Good point, it should have been a fail ofc. Thanks for pointing it out.